src/lammpsio/snapshot.py

```diff
@@ -10,20 +10,49 @@
 class Snapshot:
     """Particle configuration.
 
+    A `Snapshot` holds the data for `N` particles, the simulation `Box`, and the
+    timestep.
+
     Parameters
     ----------
     N : int
         Number of particles in configuration.
-    box : :class:`Box`
+    box : `Box`
         Simulation box.
     step : int
-        Simulation time step counter. Default of ``None`` means
-        time step is not specified.
+        Simulation time step. Default of ``None`` means time step is not
+        specified.
+    num_types : int
+        Number of particle types. If ``None``, the number of types is deduced
+        from `typeid`.
 
-    Attributes
-    ----------
-    step : int
-        Simulation time step counter.
+    Example
+    -------
+    Here is a 3-particle configuration in an triclinic box centered at the
+    origin at step 10:
+
+    .. code-block:: python
+
+        box = lammpsio.Box([-5.0, -10.0, 0.0], [1.0, 10.0, 8.0], [1.0, -2.0,
+        0.5])
+
+        snapshot = lammpsio.Snapshot(3, box, 10, num_types=None)
```

Comment:
Don't mass arguments that are defaulted, but do use keyword arguments:
```suggestion
        snapshot = lammpsio.Snapshot(N=3, box=box, step=10)
```